tools/tester.nim

```diff
@@ -414,14 +416,33 @@ proc runTest(desc: RunnerDesc, file: string): bool =
           discard "ignored; only exists for informative purposes"
         of "knownIssue":
           spec.knownIssue = some evt.value
-        of "output":
-          spec.expected.add:
-            OutputSpec(fromFile: false,
-                       output: strip(evt.value, leading=true, trailing=false))
         of "arguments":
           spec.arguments = split(evt.value, ' ')
         of "reject":
           spec.reject = parseBool(evt.value)
+        elif evt.key.startsWith("output"):
+          let output =
+            OutputSpec(fromFile: false,
+                       output: strip(evt.value, leading=true, trailing=false))
+          if evt.key == "output":
+            if usesPositional:
+              echo "cannot mix positional and non-positional 'output'"
+              return false
+            spec.expected.add output
+          else:
+            let i = find(evt.key, '.')
+            if i == -1:
```

Comment:
should this also check to see if '.' doesn't appear as the last character, as that would also be illformed?

Reply:
That's already taken care of indirectly, via the following `parseInt` call. If the string is empty or contains non-digit characters, an exception will be raised.

Reply:
Yup, I did see that, I should have mentioned it, I just meant _if_ you wanted to opt for a more specific error message, but all this is a soft suggestion.

I should add the benefit of the exception is that it's still precise, even if a little indirect.

Reply:
I'll add dedicated handling for the missing number case.